modules/dnn/src/layers/gather_elements_layer.cpp

```diff
@@ -0,0 +1,141 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "../precomp.hpp"
+#include "layers_common.hpp"
+
+
+namespace cv { namespace dnn {
+
+class GatherElementsLayerImpl CV_FINAL : public GatherElementsLayer
+{
+public:
+    GatherElementsLayerImpl(const LayerParams& params)
+    {
+        setParamsFrom(params);
+        axis = params.get<int>("axis", 0);
+    }
+
+    virtual bool supportBackend(int backendId) CV_OVERRIDE
+    {
+        return backendId == DNN_BACKEND_OPENCV;
+    }
+
+    virtual bool getMemoryShapes(const std::vector<MatShape> &inputs,
+                                 const int requiredOutputs,
+                                 std::vector<MatShape> &outputs,
+                                 std::vector<MatShape> &internals) const CV_OVERRIDE
+    {
+        CV_CheckEQ(inputs.size(), 2ull, "GatherElements: requires two inputs ");
+        outputs.assign(1, inputs[1]); // shape of output is same as indices
+        return false;
+    }
+
+    void forward(InputArrayOfArrays inputs_arr, OutputArrayOfArrays outputs_arr, OutputArrayOfArrays internals_arr) CV_OVERRIDE
+    {
+        CV_TRACE_FUNCTION();
+        CV_TRACE_ARG_VALUE(name, "name", name.c_str());
+
+        std::vector<Mat> inputs, outputs;
+        inputs_arr.getMatVector(inputs);
+        outputs_arr.getMatVector(outputs);
+
+        const Mat& data = inputs[0];
+        const Mat& indices = inputs[1];
+        Mat& out = outputs[0];
+
+        typeDispatch(outputs[0].type(), data, indices, out);
+    }
+
+    template<typename T>
+    void forward_impl(const Mat& data, const Mat& indices,  Mat& out)
```

Comment:
We need to refactor this implementation because it is just tooooooo slow. Perf with `GatherElements(data:FLOAT[2700, 1, 2914], indices:INT64[2700, 1, 81])`:
[ort_benchmark.zip](https://github.com/opencv/opencv/files/12837145/ort_benchmark.zip)

| this impl | ort |
| - | - |
| 15415.07 ms | 0.15ms |

Reply:
Improvement:

Previous impl by Aser: 15415.07 ms
After removing meaningless for loop: 1ms
Speedup: 0.49ms
ort: 0.15 ms